internal/rancher/cluster.go

```diff
@@ -40,43 +27,21 @@ type ClusterStatus struct {
 
 // ToUnstructured converts a Cluster to an unstructured object.
 func (r *Cluster) ToUnstructured() (*unstructured.Unstructured, error) {
-	rancherClusterUnstructured := &unstructured.Unstructured{}
-
-	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
-		TagName:    "json",
-		Result:     &rancherClusterUnstructured.Object,
-		DecodeHook: stringToTimeHook,
-	})
+	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(r)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create decoder: %w", err)
-	}
-
-	if err := decoder.Decode(r); err != nil {
-		return nil, fmt.Errorf("failed to decode rancher cluster: %w", err)
+		return nil, fmt.Errorf("failed to decode cluster: %w", err)
 	}
 
+	rancherClusterUnstructured := &unstructured.Unstructured{}
+	rancherClusterUnstructured.SetUnstructuredContent(obj)
 	rancherClusterUnstructured.SetGroupVersionKind(gvkRancherCluster)
-	rancherClusterUnstructured.SetCreationTimestamp(metav1.Now())
```

Comment:
curious, why is this not needed anymore?

Reply:
I had some issues previously with missing creation timestamp when converting from unstructured, I guessed it's fixed now when using `runtime.DefaultUnstructuredConverter`